### PR TITLE
Toolkit Task Improvements

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

### Changed
- Updated `ToolkitTask` system prompt to no longer mention `memory_name` and `artifact_namespace`.
- Models in `ToolkitTask` with native tool calling no longer need to provide their final answer as `Answer:`.
### Fixed
- Models occasionally hallucinating `memory_name` and `artifact_namespace` into Tool schemas when using `ToolkitTask`.
- Models occasionally providing overly succinct final answers when using `ToolkitTask`.

## Issue ticket number and link
https://discord.com/channels/1096466116672487547/1105169591619027055/1296270653238022226

----
📚 Documentation preview 📚: https://griptape--1268.org.readthedocs.build//1268/

### CODE REVIEW

File: griptape/tasks/toolkit_task.py [hunk 1]
```diff
@@ -180,9 +180,6 @@ def run(self) -> BaseArtifact:
             if subtask.output is None:
                 if len(self.subtasks) >= self.max_subtasks:
                     subtask.output = ErrorArtifact(f"Exceeded tool limit of {self.max_subtasks} subtasks per task")
-                elif not subtask.actions:
-                    # handle case when the LLM failed to follow the ReAct prompt and didn't return a proper action
-                    subtask.output = subtask.input
```
Comment (collindutter) [inline griptape/tasks/toolkit_task.py:185]: We need to handle this slightly differently depending on whether we're using native tool calling or not. Moved this logic to `ActionsSubtask`.

File: griptape/tasks/actions_subtask.py [hunk 2]
```diff
@@ -243,23 +248,29 @@ def __init_from_artifacts(self, artifacts: ListArtifact) -> None:
...
 
-            self.actions = [self.__process_action_object(action_object) for action_object in actions_list]
+            return [self.__process_action_object(action_object) for action_object in actions_list]
         except json.JSONDecodeError as e:
             logger.exception("Subtask %s\nInvalid actions JSON: %s", self.origin_task.id, e)
 
             self.output = ErrorArtifact(f"Actions JSON decoding error: {e}", exception=e)
 
+            return []
+
```
Comment (collindutter) [inline griptape/tasks/actions_subtask.py:273]: Fewer side effects, more consistent with native tool calling action parsing.

File: docs/griptape-framework/structures/rulesets.md [hunk 3]
```diff
@@ -29,9 +29,6 @@ A [Ruleset](../../reference/griptape/rules/ruleset.md) can be used to define [Ru
 [JsonSchemaRule](../../reference/griptape/rules/json_schema_rule.md)s defines a structured format for the LLM's output by providing a JSON schema.
 This is particularly useful when you need the LLM to return well-formed data, such as JSON objects, with specific fields and data types.
 
-!!! warning
-    `JsonSchemaRule` may break [ToolkitTask](../structures/tasks.md#toolkit-task) which relies on a specific [output token](https://github.com/griptape-ai/griptape/blob/e6a04c7b88cf9fa5d6bcf4c833ffebfab89a3258/griptape/tasks/toolkit_task.py#L28).
-
```
Comment (collindutter) [inline docs/griptape-framework/structures/rulesets.md:34]: Technically this is still true for non native tool calling, but that feels edge case enough where it's not worth mentioning.

File: griptape/templates/tasks/toolkit_task/system.j2 [hunk 4]
```diff
@@ -8,16 +8,14 @@ Actions: <JSON array of actions that MUST follow this schema: {{ actions_schema
...
 {% endif %}
 Repeat executing actions as many times as you need.
 If an action's output contains an error, you MUST ALWAYS try to fix the error by executing another action. 
-You must use the following format when providing your final answer:
-Answer: <final answer>
 
 Be truthful. ALWAYS be proactive and NEVER ask the user for more information input. Keep using actions until you have your final answer.
 NEVER make up actions, action names, or action paths. NEVER make up facts. NEVER reference tags in other action input values.
-
-Actions might store their output in memory as artifacts (with `memory_name` and `artifact_namespace`). If action output is stored in memory, ALWAYS try to pass it to another action. NEVER make up memory names or artifact namespaces.
```
Comment (collindutter) [inline griptape/templates/tasks/toolkit_task/system.j2:20]: This was causing the LLM to hallucinate these keys into the Tool schema. The Tool schema is descriptive enough, we don't need to mention them here.

**Review by cjkindel — DISMISSED** [review]
Code looks good to me. Personal structures/tool use cases are happy with this branch as well.

**Review by cjkindel — APPROVED** [review]

**Pull request merged**